service/history/queue/timer_queue_processor_base.go

```diff
@@ -176,7 +176,10 @@ func (t *timerQueueProcessorBase) Start() {
 
 func (t *timerQueueProcessorBase) Stop() {
 	if !atomic.CompareAndSwapInt32(&t.status, common.DaemonStatusStarted, common.DaemonStatusStopped) {
-		return
+		// initialized should also be stopped to stop TimerGate
+		if !atomic.CompareAndSwapInt32(&t.status, common.DaemonStatusInitialized, common.DaemonStatusStopped) {
```

Comment:
This is a good catch but instead of solving problem here we can fix TimerGate itself. It should have Start/Stop functions and shouldn't create any goroutine in constructor. That would be cleaner solution I think.

Reply:
I still think the code is buggy since lifecycle of timer gate is not handled correctly here.
* if you just initialize the timer queue processor without starting it, timer gate needs to be stopped separately
* if you start timer queue processor and stop it, timer gate is closed. 

 But to make minimum change. I'll handle this in the test case. The remaining concern is whether we do create timer gate without close it in the first scenario. But it should be addressed separately. I'll just add a comment.